nixos/modules/services/web-apps/weblate.nix

```diff
@@ -0,0 +1,398 @@
+        "ENGINE": "django.db.backends.postgresql",
+        "HOST": "/run/postgresql",
+        "NAME": "weblate",
+        "USER": "weblate",
+        "PASSWORD": "",
+        "PORT": ""
+      }
+    }
+
+    with open("${cfg.djangoSecretKeyFile}") as f:
+      SECRET_KEY = f.read().rstrip("\n")
+
+    CACHES = {
+      "default": {
+        "BACKEND": "django_redis.cache.RedisCache",
+        "LOCATION": "unix://${config.services.redis.servers.weblate.unixSocket}",
+        "OPTIONS": {
+            "CLIENT_CLASS": "django_redis.client.DefaultClient",
+            "PARSER_CLASS": "redis.connection.HiredisParser",
+            "PASSWORD": None,
+            "CONNECTION_POOL_KWARGS": {},
+        },
+        "KEY_PREFIX": "weblate",
+      },
+      "avatar": {
+        "BACKEND": "django.core.cache.backends.filebased.FileBasedCache",
+        "LOCATION": "/var/lib/weblate/avatar-cache",
+        "TIMEOUT": 86400,
+        "OPTIONS": {"MAX_ENTRIES": 1000},
+      }
+    }
+
+    ADMINS = (("Weblate Admin", "${cfg.smtp.user}"),)
+
+    EMAIL_HOST = "${cfg.smtp.host}"
+    EMAIL_USE_TLS = True
+    EMAIL_HOST_USER = "${cfg.smtp.user}"
+    SERVER_EMAIL = "${cfg.smtp.user}"
+    DEFAULT_FROM_EMAIL = "${cfg.smtp.user}"
+    EMAIL_PORT = 587
+    with open("${cfg.smtp.passwordFile}") as f:
+      EMAIL_HOST_PASSWORD = f.read().rstrip("\n")
+
+    CELERY_TASK_ALWAYS_EAGER = False
+    CELERY_BROKER_URL = "redis+socket://${config.services.redis.servers.weblate.unixSocket}"
+    CELERY_RESULT_BACKEND = CELERY_BROKER_URL
+
+    ${cfg.extraConfig}
+  '';
+  settings_py = pkgs.runCommand "weblate_settings.py" { } ''
+    mkdir -p $out
+    cat ${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/settings_example.py > $out/settings.py
+    cat >> $out/settings.py <<EOF${weblateConfig}EOF
+  '';
+  uwsgiConfig.uwsgi = {
+    type = "normal";
+    plugins = [ "python3" ];
+    master = true;
+    socket = "/run/weblate.socket";
+    die-on-idle = true;
+    die-on-term = true;
+    idle = 600;
+    manage-script-name = true;
+    cheap = true;
+    chmod-socket = "770";
+    chown-socket = "weblate:weblate";
+    uid = "weblate";
+    gid = "weblate";
+    wsgi-file = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/wsgi.py";
+    pyhome = pkgs.weblate;
+
+    # Some more recommendations by upstream:
+    # https://docs.weblate.org/en/latest/admin/install.html#sample-configuration-for-nginx-and-uwsgi
+    buffer-size = 8192;
+    reload-on-rss = 250;
+    workers = 8;
+    enable-threads = true;
+    close-on-exec = true;
+    umask = "0022";
+    ignore-sigpipe = true;
+    ignore-write-errors = true;
+    disable-write-exception = true;
+  };
+  environment = {
+    PYTHONPATH = "${settings_py}";
+    DJANGO_SETTINGS_MODULE = "settings";
+    GI_TYPELIB_PATH = "${pkgs.pango.out}/lib/girepository-1.0:${pkgs.harfbuzz}/lib/girepository-1.0";
+  };
+  weblate-env = pkgs.writeShellScriptBin
+    "weblate-env"
```

Comment:
It would be nice to expose the `django-admin` stuff so that someone can do `weblate-admin createsuperuser` or whatever.

Reply:
Yeah you would use it whenever you have to run admin tasks, as I see no other way to put the correct `settings.py` into the PYTHONPATH.

Reply:
Maybe make `settings.py` in the package load `/etc/weblate/settings.py` or something?

Reply:
What would be a nice thing in the future would be to have a "package with a setting", i.e. `cfg.weblate.withSettings (...)` and the settings would be baked into it.

Reply:
But why would one want to run the package outside NixOS in the first place?
Upstreams intention seems to be, that you download the git repository on your server, run a few install scripts and then edit the settings.py by hand to your likings. Before you edit the file, their software isn't usable as well. I don't see why we should offer a way to use Weblate outside of a deployment framework, when upstream doesn't.

Reply:
Precisely to be able to do admin tasks when needed, without having to use weird `weblate-env` runner. That is also a thing going out of way of upstream but a one much less convenient to use.

Reply:
> But why would one want to run the package outside NixOS in the first place? Upstreams intention seems to be, that you download the git repository on your server, run a few install scripts and then edit the settings.py by hand to your likings. Before you edit the file, their software isn't usable as well. I don't see why we should offer a way to use Weblate outside of a deployment framework, when upstream doesn't.

As said, no strong feeling on this. It helps a lot to manage multi-instances, diagnose existing setups, etc.
It can be done in a later PR if anyone has a need for it.